src/types/ast.rs

```diff
@@ -0,0 +1,399 @@
+// // Reference: https://docs.python.org/3.9/library/ast.html
+// // - https://www.cs.princeton.edu/~appel/papers/asdl97.pdf
+
+pub struct Attributes {
+    lineno: i32,
+    col_offset: i32,
+    end_lineno: Option<i32>,
+    end_col_offset: Option<i32>
+}
+
+// --------------
+
+pub enum Mod {
+    Module {
+        body: Vec<Stmt>,
+        type_ignores: Vec<TypeIgnore>
+    },
+    Interactive {
+        body: Vec<Stmt>
+    },
+    Expression {
+        body: Expr,
+    },
+    FunctionType {
+        argtypes: Vec<Expr>,
+        returns: Expr
+    }
+}
+
+// --------------
+
+// ---- #[derive(FromPyObject)]
+pub struct Stmt {
+    // TODO: Not 100% sure why this Box is needed
```

Comment:
I believe `Box` data is stored on the heap - might be worth trying to remove this